src/vm.ts

```diff
@@ -24,14 +24,52 @@ const dstColumn: String = columns[i];
 i++;
 const executionColumn: String = columns[i];
 i++;
+
+type builtins = {
+  output: string;
+  pedersen: string;
+  range_check: string;
+  ecdsa: string;
+  bitwise: string;
+  ec_op: string;
+  keccak: string;
+  poseidon: string;
+};
+
+const builtins = {
+  output: null,
+  pedersen: null,
+  range_check: null,
+  ecdsa: null,
+  bitwise: null,
+  ec_op: null,
+  keccak: null,
+  poseidon: null,
+};
+
 const program: any[][] = programSheet.getRange("A2:A").getValues();
 
+function initialize_builtins(): void {
+  let counter: number = 0;
+  const executionColumnOffset: number = columns.indexOf(executionColumn) + 1;
+  const keys: string[] = Object.keys(builtins);
+
+  for (var key of keys) {
+    builtins[key] = columns[counter + executionColumnOffset];
+    counter++;
+  }
+  runSheet
+    .getRange(`${builtins[keys[0]]}1:${builtins[keys[keys.length - 1]]}1`)
+    .setValues([keys]);
+}
+
 function step(n: number = 0): void {
   runSheet
     .getRange(`${pcColumn}1:${executionColumn}1`)
     .setValues([
       ["PC", "FP", "AP", "Opcode", "Op0", "Op1", "Res", "Dst", "Execution"],
     ]);
+  initialize_builtins();
```

Comment:
just remove this as the initialization should be made only once before any step.
put this instead in the `runUntilPc` which will be progressively filled to become a full run funtion

Reply:
Thanks for the pointer. I have moved the call to runUntilPc method now.